jcl/src/openj9.sharedclasses/share/classes/com/ibm/oti/shared/SharedClassURLClasspathHelperImpl.java

```diff
@@ -345,6 +353,7 @@ public synchronized void setClasspath(URL[] newClasspath) throws CannotSetClassp
 		if (changeMade) {
 			/*[MSG "K05b2", "setClasspath() updated classpath. Now urlCount={0}"]*/
 			printVerboseInfo(Msg.getString("K05b2", urlCount)); //$NON-NLS-1$
+			notifyClasspathChange3(id, loader, urls, urlCount, true);
```

Comment:
notifyClasspathChange2() frees the classloader->classPathEntries, undoing what notifyClasspathChange3() creates. notifyClasspathChange2() should be called first.

Reply:
Already changed as discussed.